### PR TITLE
Joint Subsystem -> core

### PR DESCRIPTION
**Pull request by drorta**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+	@Override
+	protected void subsystemPeriodic() {
+		updateInputs();
+	}
+
+	private void updateInputs() {
+		for (InputSignal<?> signal : otherSignals) {
+			motor.updateSignals(signal);
+		}
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java]: be better

File: src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java:1]: change angle -> position

File: src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package frc.robot.subsystems.jointSubsystem;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.hardware.motor.ControllableMotor;
+import frc.robot.hardware.request.IRequest;
+import frc.robot.hardware.signal.InputSignal;
+import frc.robot.subsystems.GBSubsystem;
+
+public class JointSubsystem extends GBSubsystem {
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java]:
```suggestion
public class Joint extends GBSubsystem {
```

File: src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	public JointSubsystem(
+		String logPath,
+		ControllableMotor motor,
+		IRequest<Rotation2d> positionRequest,
+		InputSignal<Rotation2d> positionSignal,
+		InputSignal... otherSignals
+	) {
+		super(logPath);
+		this.motor = motor;
+		this.commandsBuilder = new JointCommandsBuilder(this);
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java]: create it last

File: src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java:1]: missing function

File: src/main/java/frc/robot/subsystems/jointSubsystem/JointCommandsBuilder.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.RunCommand;
+
+import java.util.function.DoubleSupplier;
+
+public class JointCommandsBuilder {
+
+	private final JointSubsystem jointSubsystem;
+
+	public JointCommandsBuilder(JointSubsystem jointSubsystem) {
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/jointSubsystem/JointCommandsBuilder.java]: missing commands

File: src/main/java/frc/robot/subsystems/jointSubsystem/JointCommandsBuilder.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+		this.jointSubsystem = jointSubsystem;
+	}
+
+	public Command setPower(double power) {
+		return new RunCommand(() -> jointSubsystem.setPower(power), jointSubsystem).withName("Set power");
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/jointSubsystem/JointCommandsBuilder.java]:
```suggestion
		return new RunCommand(() -> jointSubsystem.setPower(power), jointSubsystem).withName("Set power: " + power);
```

File: src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+	private final IRequest<Rotation2d> positionRequest;
+
+	public JointSubsystem(
+		String logPath,
+		ControllableMotor motor,
+		IRequest<Rotation2d> positionRequest,
+		InputSignal<Rotation2d> positionSignal,
+		InputSignal... otherSignals
+	) {
+		super(logPath);
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java]: consider adding postion into the other signals and refresh all at once (for your choice to do it or not)

Comment (drorta) [reply src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java]: yeah but i specifically need the position for stayinplave and isatposition

Comment (YHerm) [reply src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java]: u can save it in both palces

Comment (drorta) [reply src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java]: wdym?

Comment (YHerm) [reply src/main/java/frc/robot/subsystems/jointSubsystem/JointSubsystem.java]: save it as seperate signal but also in the array

**Pull request closed without merging**